src/rgw/rgw_rest.cc

```diff
@@ -1379,10 +1379,6 @@ int RGWPostObj_ObjStore::verify_params()
   if (!s->length) {
     return -ERR_LENGTH_REQUIRED;
   }
-  off_t len = atoll(s->length);
-  if (len > (off_t)(s->cct->_conf->rgw_max_put_size)) {
-    return -ERR_TOO_LARGE;
-  }
```

Comment:
ok, so we remove this check, limit `max_len` to `rgw_max_put_size`, then rely on this `max_len` check in `RGWPostObj::execute()`:
```c++
      if (ofs > max_len) {                                                     
        op_ret = -ERR_TOO_LARGE;                                               
        return;                                                                
      }
```
the problem i see with this approach is that it doesn't detect the error until we've already written out 5GB of data. the existing check, although inaccurate, is must cheaper to enforce

more generally, the purpose of `rgw_max_put_size` isn't to limit how big an s3 object can be. instead, it's a limit on how much data the client can send in a single http request. this puts an upper bound on the cost of recovery, when the client has to retransmit requests after a timeout or retryable error. from this perspective, it doesn't matter whether the request body is all object data, or a mix of object data and form fields